libs/config/src/v6.rs

```diff
@@ -176,38 +183,163 @@ pub const fn is_unicast_link_local(ip: &Ipv6Addr) -> bool {
     (ip.segments()[0] & 0xffc0) == 0xfe80
 }
 
+pub fn generate_duid_from_config(
+    server_id: &ServerDuid,
+    link_layer_address: Ipv6Addr,
+) -> Result<Duid> {
+    match server_id.duid_type {
+        wire::v6::DuidType::LLT | wire::v6::DuidType::LL => {
+            let htype = match server_id.htype {
+                None | Some(0) => HType::Eth,
+                Some(htype_no) => {
+                    let htype_u8 = htype_no as u8; //TODO: a compromise of v4 HType
+                    HType::from(htype_u8)
+                }
+            };
+            let identifier = match &server_id.identifier {
+                None => link_layer_address,
+                Some(identifier_string) => {
+                    if identifier_string.is_empty() {
+                        link_layer_address
+                    } else {
+                        Ipv6Addr::from_str(identifier_string.as_str())
+                            .context("should be a valid ipv6 address")?
+                    }
+                }
+            };
+            if server_id.duid_type == wire::v6::DuidType::LLT {
+                let time = match server_id.time {
+                    None | Some(0) => SystemTime::now()
+                        .duration_since(SystemTime::UNIX_EPOCH)
+                        .context("unable to get system time")?
+                        .as_secs() as u32,
+                    Some(time) => time,
+                };
+                println!(
+                    "htype: {:?}, time: {:?}, identifier: {:?}",
+                    htype, time, identifier
+                );
+                Ok(Duid::link_layer_time(htype, time, identifier))
+            } else {
+                Ok(Duid::link_layer(htype, identifier))
+            }
+        }
+        wire::v6::DuidType::EN => {
+            let enterprise_id = match server_id.enterprise_id {
+                None | Some(0) => 1, //TODO: harewire to 1 temporarily
+                Some(enterprise_id) => enterprise_id,
+            };
+            let identifier = match &server_id.identifier {
+                None => {
+                    //randomly generate a 6-byte long identifier. use rand library
+                    generate_random_bytes(6)
+                }
+                Some(identifier_string) => {
```

Comment:
Pardon my pedantry, but I prefer not to include the type as a postfix in the variable name in most cases, if we could remove `_string` or `_file` from bindings and fns

Reply:
In new code this postfix is removed.